experimental/packages/opentelemetry-instrumentation/hook.mjs

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2.0 License.
+//
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021 Datadog, Inc.
+
+import { createHook } from './hook.js';
+
+const { load, resolve, getFormat, getSource } = await createHook(import.meta);
+
+export { load, resolve, getFormat, getSource };
```

Comment:
This may be a dumb question (could be that I'm missing some knowledge about how these hooks work), but why did we decide against just re-exporting like

```
import { load, resolve, getFormat, getSource } from 'import-in-the-middle/hook.mjs';
export { load, resolve, getFormat, getSource };
```

:thinking:

Reply:
This is the copied `hook.mjs` file from `import-in-the-middle`. The option to re-export `import-in-the-middle` functions didn't work because of the way the index file imports and mutates objects from another file in `import-in-the-middle`. When I tried it, the modules didn't get intercepted by iitm.

Reply:
Hmm, interesting. :thinking: 

Could it be that I'm missing some step to provoke this behaviour on my testing setup? :thinking: I tried re-exporting like above locally (and deleting `hook.js`), and it looks like the tests still pass when doing so - also, the traces generated by the example look like I would expect them too (`GET` span created, attributes added by instrumentation).

Reply:
You're right this does work! Thanks a bunch for testing that, it looks like the hook file has changed in v1.3.5 to support this so I'm going to change this to just re-export instead of copying any of the hook file content 🙂

Reply:
Amazing, thanks for testing it again with the new version :slightly_smiling_face: